test/framework/easyconfigparser.py

```diff
@@ -180,11 +180,12 @@ def test_easyconfig_constants(self):
         system_constant = constants.pop('SYSTEM')
         self.assertEqual(system_constant, {'name': 'system', 'version': 'system'})
 
-        # make sure both keys and values are only strings
+        # make sure both keys and values are of appropriate types
         for constant_name in constants:
             self.assertTrue(isinstance(constant_name, string_type), "Constant name %s is a string" % constant_name)
             val = constants[constant_name]
-            self.assertTrue(isinstance(val, (string_type, dict, tuple)), "Constant value %s is string, dict or tuple" % val)
+            self.assertTrue(isinstance(val, (string_type, dict, tuple)),
+                            "The constant %s has an acceptable type" % constant_name)
```

Comment:
@terjekv Do you mind rewording this to `should have` rather than `has`, which leads to less confusion w.r.t. failing tests (see #3231).

Also, you can still print the value, just wrap it in `str(...)` to avoid problems with tuple values:

```python
fail_msg = "The constant %s should have an acceptable type, found %s (%s)" % (constant_name, type(val), str(val))
self.assertTrue(isinstance(val, (string_type, dict, tuple)), fail_msg)

Reply:
Sorted in https://github.com/easybuilders/easybuild-framework/pull/3309/commits/3b193743172636ebf398dd7db7ee69a8638c8606